cmd/ls/ls.go

```diff
@@ -1,36 +1,57 @@
 package main
 
 import (
+	"flag"
 	"fmt"
 	"io/ioutil"
 	"os"
 )
 
-func ls(path string) {
+type parameters struct {
+	list          bool
+	humanReadable bool
+	hidden        bool
+}
+
+func ls(path string, params parameters) {
 	files, err := ioutil.ReadDir(path)
 
 	if err != nil {
 		fmt.Fprintf(os.Stderr, "Failed to read dir: %s\n", err.Error())
 		os.Exit(1)
 	}
 
-	for _, f := range files {
-		fmt.Fprintf(os.Stdout, "%-20s", f.Name())
+	if params.list {
+		printFileList(files)
+	} else {
+		printFileNames(files)
+	}
+}
+
+func printFileList(filesInfo []os.FileInfo) {
+	for _, f := range filesInfo {
+		fmt.Fprintf(os.Stdout, "%-10s %-5s %s ", f.Mode(), f.Size(), f.Name())
+		fmt.Fprintf(os.Stdout, "\n")
+	}
+	fmt.Fprintf(os.Stdout, "\n")
+}
+
+func printFileNames(filesInfo []os.FileInfo) {
+	for _, f := range filesInfo {
+		fmt.Fprintf(os.Stdout, "%s ", f.Name())
 	}
 	fmt.Fprintf(os.Stdout, "\n")
 }
 
 func main() {
 	path := "."
 
-	if len(os.Args) > 1 {
-		if os.Args[1][0] != '-' {
-			path = os.Args[1]
-		} else {
-			fmt.Fprintf(os.Stderr, "%s not implemented\n", os.Args[1])
-			os.Exit(1)
-		}
-	}
+	l := flag.Bool("l", true, "use a long listing format")
+	h := flag.Bool("h", true, "print human readable sizes")
```

Comment:
According to #3 the only parameter required is "-l". Our `ls` will be for humans by default (alien will get upset).
:-)